mt-kahypar/datastructures/sparse_map.h

```diff
@@ -639,5 +637,155 @@ struct EmptyStruct { };
 
 template<typename Key> using DynamicSparseSet = DynamicSparseMap<Key, EmptyStruct>;
 
+template <typename Key = Mandatory,
+          typename Value = Mandatory>
+class DynamicFlatMap {
```

Comment:
The DynamicSparseMap and DynamicFlatMap look very similar and share a lot of code. Can we have a common base class for both (DynamicMapBase) and implement the derived classes with CRTP pattern (see SparseMapBase and SparseMap)?

Reply:
I have implemented a common base class now and also updated the tests accordingly